src/sentry/workflow_engine/processors/workflow.py

```diff
@@ -224,7 +226,10 @@ def process_workflows(event_data: WorkflowEventData) -> set[Workflow]:
             (Q(environment_id=None) | Q(environment_id=environment.id)),
             detectorworkflow__detector_id=detector.id,
             enabled=True,
-        ).distinct()
+        )
+        .select_related("when_condition_group")
+        .prefetch_related("when_condition_group__conditions")
```

Comment:
:woohoo:!! just want to confirm that this doesn't need updating because we were doing stuff like `when_condition_group.conditions` etc after this?

Reply:
you were right 😭 we don't do `when_condition_group.conditions`. but i wrote tests with a count of the number of queries